deepcell_toolbox/utils.py

```diff
@@ -450,3 +450,265 @@ def untile_image(tiles, tiles_info, power=2, **kwargs):
+                    else:
+                        overlap_z = (tile_size_z - stride_z, tile_size_z - stride_z)
+
+                    tiles[counter] = image[b, z_start:z_end, x_start:x_end, y_start:y_end, :]
+                    batches.append(b)
+                    x_starts.append(x_start)
+                    x_ends.append(x_end)
+                    y_starts.append(y_start)
+                    y_ends.append(y_end)
+                    z_starts.append(z_start)
+                    z_ends.append(z_end)
+                    overlaps_x.append(overlap_x)
+                    overlaps_y.append(overlap_y)
+                    overlaps_z.append(overlap_z)
+                    counter += 1
+
+    tiles_info = {}
+    tiles_info['batches'] = batches
+    tiles_info['x_starts'] = x_starts
+    tiles_info['x_ends'] = x_ends
+    tiles_info['y_starts'] = y_starts
+    tiles_info['y_ends'] = y_ends
+    tiles_info['z_starts'] = z_starts
+    tiles_info['z_ends'] = z_ends
+    tiles_info['overlaps_x'] = overlaps_x
+    tiles_info['overlaps_y'] = overlaps_y
+    tiles_info['overlaps_z'] = overlaps_z
+    tiles_info['stride_x'] = stride_x
+    tiles_info['stride_y'] = stride_y
+    tiles_info['stride_z'] = stride_z
+    tiles_info['tile_size_x'] = tile_size_x
+    tiles_info['tile_size_y'] = tile_size_y
+    tiles_info['tile_size_z'] = tile_size_z
+    tiles_info['stride_ratio'] = stride_ratio
+    tiles_info['image_shape'] = image.shape
+    tiles_info['dtype'] = image.dtype
+    tiles_info['pad_x'] = pad_x
+    tiles_info['pad_y'] = pad_y
+    tiles_info['pad_z'] = pad_z
+
+    return tiles, tiles_info
+
+
+def window_3D(window_size, overlap_z=(5, 5), overlap_x=(32, 32), overlap_y=(32, 32), power=3):
+    """
+    Make a 1D window function, then infer and return a 2D window function.
+    Done with an augmentation, and self multiplication with its transpose.
+    Could be generalized to more dimensions.
+    """
+    window_z = spline_window(window_size[0], overlap_z[0], overlap_z[1], power=power)
+    window_x = spline_window(window_size[1], overlap_x[0], overlap_x[1], power=power)
+    window_y = spline_window(window_size[2], overlap_y[0], overlap_y[1], power=power)
+
+    window_z = np.expand_dims(np.expand_dims(np.expand_dims(window_z, -1), -1), -1)
+    window_x = np.expand_dims(np.expand_dims(np.expand_dims(window_x, -1), -1), -1)
+    window_y = np.expand_dims(np.expand_dims(np.expand_dims(window_y, -1), -1), -1)
+
+    window = window_z * window_x.transpose(1, 0, 2, 3) * window_y.transpose(1, 2, 0, 3)
+
+    return window
+
+
+def untile_image_3D(tiles, tiles_info, power=3, force=False, **kwargs):
+    """Untile a set of tiled images back to the original model shape.
+
+     Args:
+         tiles (numpy.array): The tiled images image to untile.
+         tiles_info (dict): Details of how the image was tiled (from tile_image).
+         power (int): The power of the window function
```

Comment:
add `force` to the docstring. Why would we set it to `True`?

Reply:
Added. And it's so we're able to still use spline interpolation outside the bounds of what is normally defined as acceptable in terms of stride_ratio/tile_size. Should normally be false but I wanted to have that option for some of the 3D stuff.